### ISSUE TITLE
Syncing error.

### ISSUE THREAD

**Issue by aktur** [cecobask/imdb-trakt-sync#62]
Hi, since few days I observe this error 
<img width="876" alt="image" src="https://github.com/user-attachments/assets/679e6d55-2142-4372-9a60-427ef88eeb51">

**Comment by cecobask** [comment]
Hi @aktur, could you please set the `IMDB_TRACE` repository secret to `true` so I can get more details?

**Comment by aktur** [comment]
<img width="876" alt="image" src="https://github.com/user-attachments/assets/38362fb6-20ce-44c9-9c6c-9852e40256f4">
...
<img width="876" alt="image" src="https://github.com/user-attachments/assets/81b5dded-d40d-43f5-9683-2047105a1cd7">

**Comment by cecobask** [comment]
It looks like the same issue I am facing when running the sync via GitHub Actions.
The interesting part is that everything works normally when I run it locally.
I can't seem to find a way to reproduce the issue locally. Could you please try locally and let me know if it works?

**Comment by paviem** [comment]
Hi. I can confirm - it works fine when run locally.

**Comment by cecobask** [comment]
Thanks for confirming, @paviem! This is quite a strange issue...
It might have something to do with the GitHub runners that the `sync` workflow gets assigned.
A potential workaround could be to set up self-hosted runner/s and experiment there.
I'm open to suggestions if somebody has a workaround.

**Comment by aktur** [comment]
Yes, indeed, locally it works normally.

**Comment by cecobask** [comment]
I'm thinking about using AWS Lambda to run part of the sync workflow.
Users will need to set up an AWS account for the required cloud infrastructure.
The workflow can be automated via Terraform. I can start working on the solution sometime next week.

The AWS free tier will cover most use cases, but even without it, the costs would be minimal.
Using the [pricing calculator](https://calculator.aws/#/createCalculator/Lambda), I got the following estimate:
```
Unit conversions

Number of requests: 2 per day * (730 hours in a month / 24 hours in a day) = 60.83 per month
Amount of memory allocated: 128 MB x 0.0009765625 GB in a MB = 0.125 GB
Amount of ephemeral storage allocated: 512 MB x 0.0009765625 GB in a MB = 0.5 GB

---

Pricing calculations

60.83 requests x 300,000 ms x 0.001 ms to sec conversion factor = 18,249.00 total compute (seconds)
0.125 GB x 18,249.00 seconds = 2,281.13 total compute (GB-s)
Tiered price for: 2,281.13 GB-s
2,281.13 GB-s x 0.0000166667 USD = 0.04 USD
Total tier cost = 0.038 USD (monthly compute charges)
Monthly compute charges: 0.04 USD
60.83 requests x 0.0000002 USD = 0.00 USD (monthly request charges)
Monthly request charges: 0.00 USD
0.50 GB - 0.5 GB (no additional charge) = 0.00 GB billable ephemeral storage per function
Monthly ephemeral storage charges: 0 USD
Lambda costs - Without Free Tier (monthly): 0.04 USD
```

**Comment by aktur** [comment]
That works for me; I already have an AWS account. If you need assistance, I can help with CloudFormation or CDK, so no third-party tools would be necessary for the infrastructure. However, could you clarify why running it inside Lambda would be beneficial? Are you concerned that the GitHub runner might not be able to handle the job?

**Comment by cecobask** [comment]
I proposed AWS Lambda because the sync workflow frequency is by default every 12 hours. Its usage will be based on demand. The duration of the workflow (happy path) for an average user with 10 lists should be around 5 minutes. There’s also the alternative of using EC2 spot instances.

I have a feeling the compute capacity or network bandwidth of the GitHub runners is not sufficient for the sync workflow. It frequently timeouts at random parts of the source code, while the same never timeout locally. I’ll have to setup a test Lambda to verify that concept.

**Comment by CheraHamza** [comment]
@cecobask I think it has to do with some problem in Trakt authorization.

![image](https://github.com/user-attachments/assets/302811d6-7498-4280-be1e-a67b728a39dd)

**Comment by cecobask** [comment]
Hi @CheraHamza, this is considered normal behaviour for the application. The log informs you that the rate limit has been reached and that it will retry the request. Usually, the second attempt after such an event is successful.

**Comment by mxkimball** [comment]
@cecobask Is there any workaround for this error? I haven't been able to successfully sync for about two weeks. Is it working for anyone?

**Comment by cecobask** [comment]
Hi @mxkimball, if you run the sync locally on your machine it works perfectly fine. I’m still investigating a potential solution for the GitHub Actions workflow.

**Comment by cecobask** [comment]
The issue has been resolved. Please sync your forks and it should work again 👍

**Comment by aktur** [comment]
Great! So, what exactly was off?

**Comment by cecobask** [comment]
> Great! So, what exactly was off?

Something strange was happening to the browser session (cookies) after the closure of the first tab. The app was spawning a new tab for most of the activities within IMDb. I couldn’t get to reproduce the issue locally. After some trial/error and debugging I found out the authentication was being removed from the session and have no idea what was causing such behaviour. I fixed it by reusing a single tab for all activities.
